src/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs

```diff
@@ -26,8 +26,10 @@ public sealed class TypeDescriptor
         // class load anyway.
         private static readonly WeakHashtable s_providerTable = new WeakHashtable();     // mapping of type or object hash to a provider list
         private static readonly Hashtable s_providerTypeTable = new Hashtable();         // A direct mapping from type to provider.
-        private static volatile Hashtable s_defaultProviders = new Hashtable(); // A table of type -> default provider to track DefaultTypeDescriptionProviderAttributes.
-        private static volatile WeakHashtable s_associationTable;
+
+        // A table of type -> default provider to track DefaultTypeDescriptionProviderAttributes.
+        private static readonly Lazy<Hashtable> s_defaultProvidersLazy = new Lazy<Collections.Hashtable>(() => new Hashtable());
```

Comment:
Yes it was null-checked [here](https://github.com/dotnet/corefx/pull/16216/files/87a2301b5d6a55d82519f8a93ea493f7bf22300c#diff-92382ea9b8df52b88206c75a0c9f5d3aL312) and I gave it more importance than init at declaration. I will revert and make it readonly.